spec/java/util/LinkedHashSet.main.lsl

```diff
@@ -124,10 +125,7 @@ automaton LinkedHashSetAutomaton
 
     proc _generateKey (visitedKeys: map<Object, Object>): Object
     {
-        result = action SYMBOLIC("java.lang.Object");
-        action ASSUME(result != null);
-        val isKeyExist: boolean = action MAP_HAS_KEY(this.storage, result);
-        action ASSUME(isKeyExist);
+        result = action MAP_GET_ANY_KEY(this.storage);
 
         val isKeyWasVisited: boolean = action MAP_HAS_KEY(visitedKeys, result);
```

Comment:
Consider using supplementary `unseen: map<Object, Object>` to store items for later retrieval as suggested by Valentin.

Reply:
Fixed